docs/standard/class-libraries-multi-targeting.md

```diff
@@ -0,0 +1,339 @@
+platform you're building fir you include different source files, for example
+like this:
+
+```xml
+  <!-- Disable default compile items -->
+
+  <PropertyGroup>
+    <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
+  </PropertyGroup>
+
+  <!-- Include all source files except the framework specific ones -->
+
+  <ItemGroup>
+    <Compile Include="**\*.cs"
+             Exclude="**\*.netstandard.cs;**\*.netfx.cs;**\*.netcore.cs" />
+  </ItemGroup>
+
+  <!-- Conditionally include framework specific source files -->
+
+  <ItemGroup Condition=" $(TargetFramework.StartsWith('netstandard')) ">
+    <Compile Include="**\*.netstandard.cs" />
+  </ItemGroup>
+  <ItemGroup Condition=" $(TargetFramework.StartsWith('net4')) ">
+    <Compile Include="**\*.netfx.cs" />
+  </ItemGroup>
+  <ItemGroup Condition=" $(TargetFramework.StartsWith('netcoreapp')) ">
+    <Compile Include="**\*.netcore.cs" />
+  </ItemGroup>
+```
+
+Instead of using `#if` you can now use `partial` classes and indicate via the
+filename which target framework it's applicable too:
+
+```csharp
+// MyClass.netstandard.cs
+public partial class MyClass
+{
+    static void SayHello()
+    {
+        Console.WriteLine("Hello from .NET Standard!");
+    }
+}
+```
+
+```csharp
+// MyClass.netfx.cs
+public partial class MyClass
+{
+    static void SayHello()
+    {
+        Console.WriteLine("Hello from .NET Framework!");
+    }
+}
+```
+
+```csharp
+// MyClass.netcore.cs
+public partial class MyClass
+{
+    static void SayHello()
+    {
+        Console.WriteLine("Hello from .NET Core");
+    }
+}
+```
+
+This is similar to what [Project Caboodle](https://github.com/xamarin/Caboodle)
+is doing. Project Caboodle provides developers essential cross-platform APIs for
+their mobile applications via a .NET Standard-based wrapper.
+
+## NuGet Packages
+
+Multi-targeting has to produce one binary per framework. If you only use project
+references within your solution you don't need to worry about that -- it's
+basically an implementation detail.
+
+But if you need to share your library outside of your solution this becomes a
+problem as consumers have to pick the correct one. To solve this problem we
+recommend that you produce a NuGet package. This will produce a single package
+containing all the binaries. Consumers simply reference the package and during
+build the appropriate binary is selected, depending on the target framework of
+the consuming project.
+
+To produce a NuGet package you need to configure your project accordingly. You
+can do this in Visual Studio by right clicking the project and selecting
+**Properties**. On the **Package** tab you need to check the box labelled
+**Generate NuGet package on build**.
+
+You can also enable package generation by editing the project file:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+  </PropertyGroup>
+
+</Project>
+```
+
+# Best practices
```

Comment:
Outside of this PR:  We really need to have a good samples that demonstrates these techniques.